### DISCUSSION TITLE
Remove all selected tree items from multiselect tree

### DISCUSSION THREAD

**Discussion by chiim** [microsoft/fluentui#29823] [Q&A] [answered]
Hi,

I've created a multiselect tree and want to add a button that removes all checked items with one click. I saw an [example in the docs](https://codesandbox.io/s/competent-thompson-xnmzgf?file=/src/example.tsx) where you can add and remove one item at a time, but that was for single-select and didn't seem applicable for my use-case. Is there a way to reset the checkedItems value on treeProps?

Here's some code to clarify my question: https://codesandbox.io/s/competent-thompson-xnmzgf?file=/src/example.tsx

**Comment by bsunderhus** [comment, marked as answer]
`useHeadlessFlatTree` allows you to pass a controlled value of `checkedItems` and a callback to when you should update that state with `onChekcedChange`, with that you can take control of the checked state. Once in control of the checked state, to reset it simply set the state to be an empty iterable (like a map, or an array for example).

Here's an updated version of your code: https://codesandbox.io/s/awesome-wescoff-rxxyhr?file=/src/example.tsx

Also, here's a [TS playground link](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAKjgQwM5wEoFNkGN4BmUEIcARFDvmQNwBQokscA3nXHAGIA2yMAKpSwAadl14ChASRhYQojoKxYZcgDLIAnhACuMBXAASOACbcsqVDz5KVskAAViYVAZ2osx5GYtWJtgH0dADtUGGQAI3MDWwBhAAssXABrLBME5GCAcywAET5kAwAhPRgIYJjpewA1ZG4dETFbAGUsc3xgctr6rDoAXzgiEnIAAQIe4JgdYAB6SjwYAFpcEkhgrEnUWjo6FdD4ZoBRNUPY-ikAeQA5AIBZC9zDuABechAdbhhgD3aYWjgZjM4Lh4pkcnAyuRUMBsuYyINoHBobCsEi2kkvuUdjBNGBUbF3GUQKpSK8vD5LNZJMoSU4IC44AAyVjA8qySYALiRMCgMKycD69F25TCcGA9lQXIJYRIJIA2gBdF5wOViFhwABudQaXLIAEYyMJWZMNjBdWosBq2nA9UbxXIbfC+gZ1Vqerq9YsDUawMhKJNujryN7jeyzeQLVbuHAAEx2+yOgUuzXarAexYxw1wX3+mCBtPBrN7MPmy3WuNihOZpNqlPu8iZo3F02lqM2+MO6vO2tuoNkGNerM50353WN0MtiNl6MV+2kA01jiu1Njr2Dn1+kcrhvriecqdtgDMHfnTuTvYL-YzQ83Ae3-aLbMnZEj5ZPsbPPfvA4HIeHd-rK9x2bfcX2nOBj0rB0F27Jc6z7H9rw3XNRx3YCn1A19o0gucP0XFkL11Q9HxNTDwNtKDSGIgU6AVIUsAADyYeA9lFVpfk6YJlQACgASheAA+VgxFY+A5RBJJUhMElXDRGAEkktIZKVV5sAWAA6dwsGacJZG4sQOD4wS4HWAB3OBbmQMAAB5bBJfMjRadEOi6VMBO4uU5X1JCISgBoFQVXixF4+gOFEwZ-CEZUtPJcxKUi5Qgn2SJzG4udZLYDgOBMLACGQD4YAuPFghkrkvJDB8jSvCqfzIBUDDCxIUiUiUGrRDjyluCAcq5I4TjOS4bnuR42vKBTmvSUFsgLbiy0mI0TAKfjniEjx5KaqSZO4xbwnUiSJpkoKOD6EKdg4QE4GaeJdG4ExgXMP0IUSYENrSSj0D9KAtBEkV4DGh6oAAQW4aNXiMlbhKygEgWwNantRfapPeiEIDgCJURCRG0gMuTxs2iVuLMiyrL407jqFDhKCmKAuP0qHrIEnGOGskoYDKLj-uAFJnhYf6cCBkG+gE2IAZQEHrJmVn2cZqHmapWxWHUpXxhsIR1JyaksDpFw+IGP1gGQRZeHR7hnjIdiMU4sgZdlpdAagL7NHUoYQG4lXNfU9K+KNN2ErsORlqEzLbay8L1RA-Q4CV9SeRpextfQAZXnduz7HVrBNdpZxUD40KQ8pjOdBpuA6fzrLrPlqoHRYaPY-9xxs4GVJNB5lOq5AdSLyFpmy9s9uNG0PQBJYCO+gl1P1C0XQYBtsu4Alyu47kWeQ7JkOTr6HuF79leJZt07BToIA) just in case the codesandbox goes down:


```ts
export const Selection = () => {
  const [checkedItems, setCheckedItems] = React.useState(
    () => new Map<TreeItemValue, TreeSelectionValue>([["1-2", true]])
  );
  const flatTree = useHeadlessFlatTree_unstable(items, {
    defaultOpenItems: ["1", "2", "2-1", "2-2"],
    checkedItems,
    selectionMode: SELECTION_MODE,
    onCheckedChange: (event, data) => setCheckedItems(data.checkedItems)
  });

  // Should clear the checked items array
  const onClearAll = () => {
    // Reset the checked items to be unchecked
    setCheckedItems(new Map());
  };

  return (
    <>
      <Button onClick={onClearAll}>Clear all</Button>
      <FlatTree {...flatTree.getTreeProps()} aria-label="Selection">
        {Array.from(flatTree.items(), (flatTreeItem) => {
          const { content, ...treeItemProps } = flatTreeItem.getTreeItemProps();
          return (
            <FlatTreeItem {...treeItemProps} key={flatTreeItem.value}>
              <TreeItemLayout>{content}</TreeItemLayout>
            </FlatTreeItem>
          );
        })}
      </FlatTree>
    </>
  );
};

```

**Reply by chiim** [reply]
Wow spot on. Thank you so much, this made my day. Is there a place where I could've read documentation about this or is that not yet available? Just want to know if I've missed out on reading about it somewhere :)

**Reply by bsunderhus** [reply]
This is definitely a spot for improvement on the `useHeadlessFlatTree` documentation part, I'll see to update the doc on this point! thx for the question :)
